rust/connlib/tunnel/src/control_protocol.rs

```diff
@@ -203,25 +206,24 @@ where
     }
 
     fn start_ice_candidate_handler(&self, conn_id: ConnId) -> Result<()> {
```

Comment:
I couldn't quite tell whether we actually need this two-phase mechanism here where we don't immediately start reading ICE candidates from the channel. If it is not needed, we could remove this function and immediately send ICE candidates to the other party.